engine/src/flutter/lib/web_ui/lib/src/engine/canvaskit/surface.dart

```diff
@@ -149,17 +149,18 @@ class Surface extends DisplayCanvas {
 
     if (browserSupportsCreateImageBitmap) {
       JSObject bitmapSource;
+      DomImageBitmap bitmap;
       if (useOffscreenCanvas) {
-        bitmapSource = _offscreenCanvas! as JSObject;
+        bitmap = _offscreenCanvas!.transferToImageBitmap();
       } else {
         bitmapSource = _canvasElement! as JSObject;
+        bitmap = await createImageBitmap(bitmapSource, (
```

Comment:
Won't this pay the cost of `createImageBitmap` even if we're rendering using WebGL?

Reply:
This branch is for the case where (1) we are rendering with one WebGL canvas but (2) we don't have support for `OffscreenCanvas`. In this case we create a single <canvas> element with a WebGL context and use that like an `OffscreenCanvas`; the canvas is never added to the DOM, it is only used for creating bitmaps for the `bitmaprenderer` canvases.